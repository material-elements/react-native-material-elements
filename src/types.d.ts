export * from './components/types';
export * from './types/index';
