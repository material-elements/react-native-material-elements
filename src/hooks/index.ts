export * from './useRestyle';
export * from './useThemedProps';
