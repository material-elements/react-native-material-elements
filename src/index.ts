export * from './components';
export * from './hooks';
export * from './libraries';
export * from './utils';
