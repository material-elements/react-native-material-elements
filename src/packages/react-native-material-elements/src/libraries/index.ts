export * from './themes';
export * from './style';
