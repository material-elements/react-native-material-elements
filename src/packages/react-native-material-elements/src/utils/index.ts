export * from './styleGenerator';
export * from './utils';
export * from './dimensions';
