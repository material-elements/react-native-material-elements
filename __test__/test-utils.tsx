import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';

export const ThemeWrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: ThemeWrapper, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
