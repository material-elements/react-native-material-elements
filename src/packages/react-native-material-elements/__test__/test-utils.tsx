import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';
import { ThemeProviderProps } from '../src/libraries/types';

interface CustomRenderOptions extends Omit<RenderOptions, 'wrapper'> {
  themeProps?: Partial<ThemeProviderProps>;
}

export const ThemeWrapper: React.FC<ThemeProviderProps> = ({ children, ...props }) => {
  return <ThemeProvider {...props}>{children}</ThemeProvider>;
};

export const customRender = (ui: React.ReactElement, options?: CustomRenderOptions) => {
  const { themeProps, ...rest } = options || {};
  return render(ui, {
    wrapper: wrapperProps => <ThemeWrapper {...themeProps}>{wrapperProps.children}</ThemeWrapper>,
    ...rest,
  });
};

export * from '@testing-library/react-native';
export { customRender as render };
