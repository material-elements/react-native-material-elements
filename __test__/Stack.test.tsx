import React from 'react';
import { render } from './test-utils';
import { Stack } from '../src';
import { StyleSheet, Text, View } from 'react-native';

describe('Stack Component', () => {
  const ref = React.createRef<View>();
  const mockStackTestId = 'mock_stack_test_id';

  it('should render correctly', () => {
    const { toJSON } = render(
      <Stack>
        <Text>Hello</Text>
      </Stack>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    const { getByText } = render(
      <Stack>
        <Text>Child 1</Text>
        <Text>Child 2</Text>
      </Stack>,
    );

    expect(getByText('Child 1')).toBeTruthy();
    expect(getByText('Child 2')).toBeTruthy();
  });

  it('should forward ref correctly', () => {
    render(
      <Stack ref={ref}>
        <View />
      </Stack>,
    );

    expect(ref.current).not.toBeNull();
    expect(ref.current).toBeInstanceOf(View);
  });

  it('should apply the styles', () => {
    const { getByTestId } = render(
      <Stack testID={mockStackTestId} style={{ backgroundColor: 'red' }}>
        <Text>Hello</Text>
      </Stack>,
    );

    const stack = getByTestId(mockStackTestId);
    expect(stack).toBeDefined();

    const flattedStyles = StyleSheet.flatten(stack.props.style);
    expect(flattedStyles.backgroundColor).toEqual('red');
  });

  it('should change the stack component direction when passed the direction prop row', () => {
    const { getByTestId } = render(
      <Stack direction="row" testID={mockStackTestId} style={{ backgroundColor: 'red' }}>
        <Text>Hello</Text>
      </Stack>,
    );

    const stack = getByTestId(mockStackTestId);
    const flattedStyles = StyleSheet.flatten(stack.props.style);
    expect(flattedStyles.display).toEqual('flex');
    expect(flattedStyles.flexDirection).toEqual('row');
  });
});
