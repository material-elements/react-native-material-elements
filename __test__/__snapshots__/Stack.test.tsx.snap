// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack Component should render correctly 1`] = `
<View
  style={
    [
      {},
      {},
      [
        {
          "maxWidth": "100%",
          "width": "100%",
        },
        false,
        undefined,
      ],
    ]
  }
>
  <View
    style={{}}
  >
    <Text>
      Hello
    </Text>
  </View>
</View>
`;
